File: Client/ProEventos-App/src/app/models/evento.ts
import { Lote } from "./lote";
import { Palestrante } from "./palestrante";
import { RedeSocial } from "./rede_social";

export interface Evento {
    id: number;
    local: string;
    dataEvento?: Date;
    tema: string;
    qtdPessoas: number;
    lote: string;
    imagemURL: string;
    telefone: string;
    email: string;
    lotes: Lote[];
    redesSociais: RedeSocial[];
    palestranteEventos: Palestrante[];
}

File: Server/src/ProEventos.Domain/Evento.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ProEventos.Domain
{
    //[Table("EventosDetalhes")] Muda o nome da tabela salva no banco de dados
    public class Evento
    {
        //Especifica a chave primária deste objeto para o EF
        //[Key]
        //public int Codigo {get; set;}
        public int Id { get; set; }
        public string Local {get; set;}
        public DateTime? DataEvento {get; set;}
        //[NotMapped] //Este campo não será criado/salvo no BD.
        //public int ContagemDias { get; set; }
        [Required]
        [MaxLength(50)]
        public string Tema {get; set;}
        public int QtdPessoas {get; set;}
        public string Lote {get; set;}
        public string ImagemURL {get; set;}
        public string Telefone { get; set; }
        public string Email { get; set; }
        public IEnumerable<Lote> Lotes { get; set; }
        public IEnumerable<RedeSocial> RedesSociais { get; set; }
        public IEnumerable<PalestranteEvento> PalestranteEventos { get; set; }
    }
}